EIPS/eip-5792.md

```diff
@@ -207,23 +169,66 @@ As with the return value of `wallet_sendCalls`, the batch identifier may be any
 }
 ```
 
-### `atomicBatch` Capability
+### `wallet_getCapabilities`
+
+An app calling the above-defined methods without prior knowledge that the wallet supports them risks degrading user experience considerably, so an additional RPC for querying the wallet's capabilities is provided.
 
-Indicates that a wallet can execute multiple calls atomically as part of a single transaction.
+In addition to, or instead of, querying the wallet directly for capabilities, the same capability objects could be expressed out-of-band (such as in the `sessionProperty.capabilities` object persisted in a CAIP-25-conformant wallet provider interface) or in a well-known location (such as a URL derived from an [EIP-6963](./eip-6963.md) `rdns` identifier). Provider abstractions may also cache capabilities from previous requests or otherwise inject them from out-of-band. If querying multiple ways leads to different results, or to override caches or provider injections, direct RPC communication with the wallet SHOULD always be taken as the canonical and current expression of capabilities.
+
+#### `wallet_getCapabilities` RPC Specification
 
-If a wallet indicates it supports the `atomicBatch` capability, it MUST submit calls submitted with `wallet_sendCalls` as part of a single transaction.
+The key of each item per chain is names a capability and the value can be of any shape. Capabilities are returned per chain because wallets may support different capabilities across chains.
 
-#### `atomicBatch` Capaiblity Specification
+We expect the community to align on the syntax and semantics of additional capabilities in separate ERCs over time.
+
+
+```typescript
+type GetCapabilitiesParams = [];
+
+type GetCapabilitiesResult = Record<`0x${string}`, <Record<string, any>>; // Hex chain id
+```
+
+##### `wallet_getCapabilities` Example Return Value
+
+The capabilities below are for illustrative purposes.
+
+```json
+{
+  "0x2105": {
+    "paymasterService": {
+      "supported": true
+    },
+    "sessionKeys": {
+      "supported": true
+    }
+  },
+  "0x14A34": {
+    "paymasterService": {
+      "supported": true
+    }
+  }
+}
+```
+
+#### `atomicBatch` Capability 
+
+Like the illustrative examples given above and other capabilities to be defined in future EIPs, the capability to execute calls delivered via the above-defined methods in a single transaction can be attested by the wallet in boolean form. 
+
+If a wallet has affirmatively expressed this `atomicBatch` capability, it MUST submit calls submitted with `wallet_sendCalls` as part of a single transaction.
+
+##### `atomicBatch` Capability Specification
 
 ```typescript
 type AtomicBatchCapability = {
   supported: true
 }
 ```
 
-##### `wallet_getCapabilities` Example Return Value
+The only valid JSON-RPC values for this capability are `true` or `false`; if a returned value for "supported" value is typed as a string or number, it SHOULD be treated as malformed.
```

Comment:
Why? What if I want a capability like "offChainFunding" and want to provide a URI where info can be fetched?

Reply:
Mmm, reading this as a diff may be obscuring the meaning (or it might be badly written) , i was referring specifically to the `atomicBatch` boolean, not to all other possible object capabilities expressable in this RPC-or-its-equivalents.  any help making this text link that otherwise arbitrary constraint to the privacy considerations below would be appreciated, but I was trying not to overload this section with explanation.

Reply:
it's also totally fine to `debate if this should be more tightly specced` in the first place, as i well appreciate that i'm kind of the OCap hardliner in this conversation and also not a co-author 😅

Reply:
Oh ok sorry, yeah got confused by the diff. That is helpful clarity.

Reply:
> i'm kind of the OCap hardliner in this conversation 

Happy to hear the steelman for more tightly specifying :)